src/Psalm/Internal/Codebase/Functions.php

```diff
@@ -548,14 +556,14 @@ public function isCallMapFunctionPure(
         );
 
         if (!$function_callable->params
-            || ($args !== null && \count($args) === 0)
+            || ($args !== null && count($args) === 0)
             || ($function_callable->return_type && $function_callable->return_type->isVoid())
         ) {
             return false;
         }
 
         $must_use = $function_id !== 'array_map'
-            || (isset($args[0]) && !$args[0]->value instanceof \PhpParser\Node\Expr\Closure);
+            || (isset($args[0]) && !$args[0]->value instanceof Closure);
```

Comment:
It would make sense to alias it to something like `ClosureNode`, to prevent confusion.

Reply:
We could for now. Especially PhpParser\* is one of those I’d like to find a way to make them shorter but also more distinguishable – and that in a uniform way throughout the code. But I’s like to make some tests before and then open a new issue to discuss my findings